services/upload/src/main/java/app/coronawarn/server/services/federation/upload/keys/DiagnosisKeyPersistenceLoader.java

```diff
@@ -25,6 +25,8 @@ public DiagnosisKeyPersistenceLoader(FederationUploadKeyService uploadKeyService
   @Override
   public List<FederationUploadKey> loadDiagnosisKeys() {
     return this.uploadKeyService
-        .getPendingUploadKeys(ExpirationPolicy.of(uploadConfig.getExpiryPolicyMinutes(), ChronoUnit.MINUTES));
+        .getPendingUploadKeys(
+            ExpirationPolicy.of(uploadConfig.getExpiryPolicyMinutes(), ChronoUnit.MINUTES),
+            this.uploadConfig.getRetentionDays());
```

Comment:
I don't get this...

Reply:
What part? If it's the second parameter, it's a safety check to load keys only younger than 14 days. To avoid sending incorrect data to EFGS. 

Since the keys in the Upload table are duplicated form Diagnosis Key Table with a DB trigger, distribution will handle the retention policy. So this check shouldn't even be needed in the first place. It's just a sanity check to avoid any wrong keys leaking into our upload batch. Plus, in our test systems, old Test Data Generation keys _could_ end up with a wrong rolling start interval before being deleted.

Reply:
@fredrb but on which date is this checked?
1. submission timestamp?
1. rolling_start_interval_number?